dev/integration_tests/flutter_gallery/test_driver/e2e_utils.dart

```diff
@@ -28,10 +28,21 @@ Future<void> watchPerformance(
     }
     return true;
   }());
+
+  // The engine could batch FrameTimings and send them only once per second.
+  // Delay for a sufficient time so either old FrameTimings are flushed and not
+  // interfering our measurements here, or new FrameTimings are all reported.
+  final Future<void> delayForFrameTimings =
+      Future<void>.delayed(const Duration(seconds: 2));
```

Comment:
This sounds potentially flaky. Is there any way to improve coordination with the engine to avoid magic delays like this?